substratevm/src/org.graalvm.polyglot.nativeapi/src/org/graalvm/polyglot/nativeapi/PolyglotNativeAPI.java

```diff
@@ -1560,6 +1559,160 @@ public static PolyglotStatus poly_close_handle_scope(PolyglotIsolateThread threa
+                    "Checks if this exception has a guest language exception object attached to it.",
+                    "",
+                    " @param exception Handle to the exception object.",
+                    " @param result The result of the check.",
+                    " @return poly_ok if everything went ok, otherwise an error occurred.",
+                    "",
+                    " @since 1.0",
+    })
+    public static PolyglotStatus poly_exception_has_object(PolyglotIsolateThread thread, PolyglotExceptionHandle exception, CBoolPointer result) {
+        return withHandledErrors(() -> {
+            PolyglotException e = fetchHandle(exception);
+            result.write(CTypeConversion.toCBoolean(e.getGuestObject() != null));
+        });
+    }
+
+    @CEntryPoint(name = "poly_exception_get_object", documentation = {
+                    "Gets the handle to the guest exception object. This object can then be used in other poly methods.",
+                    "",
+                    " @param exception Handle to the exception object.",
+                    " @param result The handle to the guest object if it exists.",
+                    " @return poly_ok if everything went ok, otherwise an error occurred.",
+                    "",
+                    " @since 1.0",
+    })
+    public static PolyglotStatus poly_exception_get_object(PolyglotIsolateThread thread, PolyglotExceptionHandle exception, PolyglotValuePointer result) {
+        return withHandledErrors(() -> {
+            PolyglotException e = fetchHandle(exception);
+            Value guestObject = e.getGuestObject();
+            if (guestObject == null) {
+                reportError("Attempted to get the guest object of an exception that did not have one.", poly_generic_failure);
+            } else {
+                result.write(createHandle(guestObject));
+            }
+        });
+    }
+
+    @CEntryPoint(name = "poly_exception_get_debug_stack_trace", documentation = {
+                    "Gets the full host and guest stack traces as a string.",
+                    "The returned string is valid until the next call to this function",
+                    "",
+                    " @param exception Handle to the exception object.",
+                    " @param buffer UTF-8 string representing the stack trace. Can be NULL.",
+                    " @param buffer_size Size of the user-supplied buffer.",
+                    " @param result If buffer is NULL, this will contain the buffer size required to put the trace string in, otherwise, it will contain the number of bytes written",
+                    " @return poly_ok if everything went ok, otherwise an error occurred.",                    "",
+                    " @since 1.0",
+    })
+    public static PolyglotStatus poly_exception_get_debug_stack_trace(PolyglotIsolateThread thread, PolyglotExceptionHandle exception, CCharPointer buffer, UnsignedWord buffer_size, SizeTPointer result) {
```

Comment:
I think we should remove this for now as I am not quite sure it should go to the API. This is more a debugging feature, and maybe we have a better way around it.

Reply:
I agree, removing it now.